### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by AndrewHudzenko**

### CODE REVIEW

File: src/main/java/mate/jdbc/dao/DriverDaoImpl.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
 public class DriverDaoImpl implements DriverDao {
     @Override
     public Driver create(Driver driver) {
-        String query = "INSERT INTO drivers (name, license_number) VALUES (?, ?)";
+        String query = "INSERT INTO drivers (name, licenseNumber) VALUES (?, ?)";
```
Comment (justpopova) [inline src/main/java/mate/jdbc/dao/DriverDaoImpl.java:20]:
```suggestion
        String query = "INSERT INTO drivers (name, license_number) VALUES (?, ?)";
```
this is the right way of giving column names, not in uppercase. If in your DB your columns are in camelCase then better change it.

**Review by justpopova — APPROVED** [review]
good job!

File: src/main/java/mate/jdbc/dao/DriverDaoImpl.java (hunk 1 above)
Comment (dSkarzh) [inline src/main/java/mate/jdbc/dao/DriverDaoImpl.java:20]:
```suggestion
        String query = "INSERT INTO drivers (name, license_number) VALUES (?, ?)";
```

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertRequest = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?);";
```
Comment (dSkarzh) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:22]:
```suggestion
        String query = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?);";
```
it is obvious from the context what this query does so you can simply call it `query`
same goes for other queries

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                        = connection.prepareStatement(updateRequest)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update a car " + car, e);
+        }
+        addDriverToCar(car);
+        removeDriverFromCar(car);
```
Comment (dSkarzh) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:107]:
```suggestion
        removeDriversFromCar(car);
        addDriversForCar(car);
```
think what should we do first? adding driver and then deleting? or vice versa?

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long carId) {
+        String deleteRequest = "UPDATE cars SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement softDeleteCarStatement
```
Comment (dSkarzh) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:115]:
```suggestion
                PreparedStatement statement
```
same thing as with query namings - obvious context

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+                        = connection.prepareStatement(getAllCarsRequest)) {
+            getAllCarsStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = parseCarWithManufacturerFromResultSet(resultSet);
+                cars.add(car);
+            }
+            for (Car currentCur : cars) {
+                currentCur.setDrivers(getDriversForCar(currentCur.getId()));
+            }
```
Comment (dSkarzh) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:143]: in this loop we have nested connection problem

**Review by dSkarzh — CHANGES_REQUESTED** [review]

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java (hunk 5 above)
Comment (AndrewHudzenko) [reply src/main/java/mate/jdbc/dao/CarDaoImpl.java:143]: idk how we can do it in another way

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+                        = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update a car " + car, e);
+        }
+        removeDriverFromCar(car);
+        addDriverToCar(car);
```
Comment (dSkarzh) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:107]:
```suggestion
        removeDriversFromCar(car);
        addDriversToCar(car);
```

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+                cars.add(car);
+            }
+            for (Car currentCur : cars) {
+                currentCur.setDrivers(getDriversForCar(currentCur.getId()));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add all cars by driver id: " + driverId, e);
+        }
+    }
```
Comment (dSkarzh) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:148]:
```suggestion
    @Override
    public List<Car> getAllByDriver(Long driverId) {
        String query = "SELECT cars.id, cars.model, cars.manufacturer_id, "
                + "manufacturers.name, manufacturers.country, "
                + "FROM cars, cars_drivers, manufacturers "
                + "WHERE cars_drivers.driver_id = ? AND cars.is_deleted = false;";
        List<Car> cars = new ArrayList<>();
        try (Connection connection = ConnectionUtil.getConnection();
                PreparedStatement statement
                        = connection.prepareStatement(query)) {
            statement.setLong(1, driverId);
            ResultSet resultSet = statement.executeQuery();
            while (resultSet.next()) {
                Car car = parseCarWithManufacturerFromResultSet(resultSet);
                cars.add(car);
            }
        } catch (SQLException e) {
            throw new DataProcessingException("Can't add all cars by driver id: " + driverId, e);
        }
        for (Car car : cars) {
            car.setDrivers(getDriversForCar(car.getId()));
        }
        return cars;
    }
```
you open connection in `getDriversForCar()`, but you already have a connection opened in `getAllByDriver()`

**Review by dSkarzh — CHANGES_REQUESTED** [review]

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Car car = parseCarWithManufacturerFromResultSet(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add all cars by driver id: " + driverId, e);
+        }
+        for (Car currentCur : cars) {
```
Comment (dSkarzh) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:144]:
```suggestion
        for (Car car : cars) {
```

**Review by dSkarzh — APPROVED** [review]

**Pull request closed without merging**